src/topojson/topojson.ml

```diff
@@ -616,50 +616,74 @@ module Make (J : Intf.Json) = struct
         Geometry.to_json g
 
   let v ?bbox topojson = { bbox; topojson }
-end
 
-module Geojson = Geojson.Make (Ezjsonm_parser)
-
-let extract_lines_from_geometry (t : Geojson.t) :
-    Geojson.Geometry.Position.t array list =
-  let open Geojson.Geometry in
-  let geojson = Geojson.geojson t in
-  match geojson with
-  | Geometry g -> (
-      let geo = geometry g in
-      match geo with
-      | LineString ls -> [ LineString.coordinates ls ]
-      | MultiLineString ml ->
-          let lines = MultiLineString.lines ml in
-          let arr = Array.map LineString.coordinates lines in
-          Array.to_list arr
-      | Polygon p ->
-          let rings = Polygon.rings p in
-          let arr = Array.map LineString.coordinates rings in
-          Array.to_list arr
-      | MultiPolygon mp ->
-          let rings = MultiPolygon.polygons mp in
-          let r = Array.to_list (Array.map Polygon.rings rings) in
-          let arr = Array.map LineString.coordinates (Array.concat r) in
-          Array.to_list arr
-      | Point _ -> []
-      | MultiPoint _ -> []
-      | _ -> [])
-  | _ -> failwith "Not a valid Geometry"
-
-let extract_from_feature (f : Geojson.Feature.t) :
-    Geojson.Geometry.Position.t array list =
-  let open Geojson.Feature in
-  let geometry = geometry f in
-  match geometry with
-  | Some t ->
-      let p = Geojson.Geometry t in
-      let geo = Geojson.v p in
-      extract_lines_from_geometry geo
-  | None -> []
-
-let extract_from_feature_collection (fc : Geojson.Feature.Collection.t) :
-    Geojson.Geometry.Position.t array list =
-  let open Geojson.Feature.Collection in
-  let features = features fc in
-  List.concat (List.map extract_from_feature features)
+  module Geojson = Geojson.Make (J)
+
+  let extract_lines_from_geometry (t : Geojson.t) :
+      Geojson.Geometry.Position.t array list =
+    let open Geojson.Geometry in
+    let geojson = Geojson.geojson t in
+    match geojson with
+    | Geometry g -> (
+        let geo = geometry g in
+        match geo with
+        | LineString ls -> [ LineString.coordinates ls ]
+        | MultiLineString ml ->
+            let lines = MultiLineString.lines ml in
+            let arr = Array.map LineString.coordinates lines in
+            Array.to_list arr
+        | Polygon p ->
+            let rings = Polygon.rings p in
```

Comment:
Hmm thinking about this a little more, perhaps we have to treat rings slightly differently because the first and last point are technically connected.